### DISCUSSION TITLE
Infer strict type for empty object

### DISCUSSION THREAD

**Discussion by Stadly** [colinhacks/zod#3148] [Q&A] [answered]
With the following code, `Empty` is inferred as `{}`. But `{}` in TypeScript matches any object, not just the empty object. Is there any way to infer a more strict type, such as `Record<string, never>`?

```ts
const empty = z.object({});
type Empty = z.infer<typeof empty>;
const myObject: Empty = {
  foo: "bar", // I want this to fail!
};

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const empty = z.record( z.never() )
type Empty = z.infer<typeof empty>
const myObject: Empty = {
    foo: 'bar',
//  ^^^
//  Type 'string' is not assignable to type 'never'.
}
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Comment by Stadly** [comment]
Perfect! Thanks :)
